common/errors.js

```diff
@@ -436,5 +440,44 @@ window.onerror = function() {
     }
 
     logError(error);
-
 };
+
+var errorVisible = false;
+var offlineErrorTemplate = function (error) {
+    var errorVisible = document.getElementById('divErrorModal');
+    if (!document || !document.body || errorVisible) return;
+
+    var errName = error.constructor.name;
+    errName = (errName.toLowerCase() !== 'error') ? errName : "Terminal Error";
+
+    var html  = '<div modal-render="true" tabindex="-1" role="dialog" class="modal fade in" index="0" animate="animate" modal-animation="true" style="z-index: 1050; display: block;">'
```

Comment:
This is almost identical to the one above it. We could create a function that would generate this for us given a set of parameters (to avoid duplicating it).